library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/api/client/QuiltArmorMaterialExtensions.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.entity.api.client;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.item.ArmorMaterial;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+@InjectedInterface(ArmorMaterial.class)
+public interface QuiltArmorMaterialExtensions {
+	@Environment(EnvType.CLIENT)
+	@NotNull Identifier getTexture();
+}
```

Comment:
I would still add this interface via a mixin to armor material because injected interfaces are weird